snapcraft/application.py

```diff
@@ -99,22 +101,42 @@ def __init__(self, *args, **kwargs) -> None:
             self._snapcraft_yaml_path: pathlib.Path | None = self._resolve_project_path(
                 None
             )
+            with self._snapcraft_yaml_path.open() as file:
+                self._snapcraft_yaml_data = util.safe_yaml_load(file)
         except FileNotFoundError:
-            self._snapcraft_yaml_path = None
+            self._snapcraft_yaml_path = self._snapcraft_yaml_data = None
+
+        self._known_craft_app_base = self._get_known_craft_app_base()
 
         for craft_var, snapcraft_var in MAPPED_ENV_VARS.items():
             if env_val := os.getenv(snapcraft_var):
                 os.environ[craft_var] = env_val
 
+    def _get_known_craft_app_base(self) -> bool:
+        if self._snapcraft_yaml_data:
+            base = self._snapcraft_yaml_data.get("base")
+            build_base = self._snapcraft_yaml_data.get("build-base")
+
+            # core22 is the only current base not handled by the craft-application codepath
+            craft_app_bases = CURRENT_BASES - {"core22"}
+
+            # we know for sure that we're handling a core24 or newer project
+            if base in craft_app_bases or build_base in craft_app_bases:
```

Comment:
playing devil's advocate here, what should be the code path for this (unlikely) scenario:

```
base: core24
build-base: core22
```
?

Reply:
that should use the non carft-application implementation as what you are asking there is to use the core22 feature set to target core24

Reply:
Good point, I am now calling `utils.get_effective_base()`, which aligns with Sergio's statement.